website/blog/2021-04-21-2.3.0.md

```diff
@@ -0,0 +1,1738 @@
+
+Regression from v2.2.0.
+
+<!-- prettier-ignore -->
+```jsx
+// Input
+a(b => c => function (){})
+
+// Prettier 2.2
+TypeError: Cannot read property 'length' of undefined
+
+// Prettier 2.3
+a((b) => (c) => function () {});
+```
+
+#### Improve formatting for inline decorators ([#10296](https://github.com/prettier/prettier/pull/10296) by [@fisker](https://github.com/fisker))
+
+<!-- prettier-ignore -->
+```jsx
+// Input
+class Foo {
+  @decorator([]) bar() {}
+  @decorator(
+    []
+  ) baz() {}
+}
+
+// Prettier 2.2
+class Foo {
+  @decorator([]) bar() {}
+  @decorator([])
+  baz() {}
+}
+
+// Prettier 2.3
+class Foo {
+  @decorator([]) bar() {}
+  @decorator([]) baz() {}
+}
+```
+
+#### Fix ASI protection for private fields ([#10334](https://github.com/prettier/prettier/pull/10334) by [@thorn0](https://github.com/thorn0))
+
+<!-- prettier-ignore -->
+```jsx
+// Input
+class C {
+  #field = 'value';
+  ["method"]() {}
+}
+
+// Prettier 2.2 (with --no-semi)
+class C {
+  #field = "value"
+  ["method"]() {}
+}
+
+// Prettier 2.3 (with --no-semi)
+class C {
+  #field = "value";
+  ["method"]() {}
+}
+```
+
+#### Support Module Blocks proposal ([#10417](https://github.com/prettier/prettier/pull/10417) by [@sosukesuzuki](https://github.com/sosukesuzuki), [@thorn0](https://github.com/thorn0))
+
+Support formatting for [Module Blocks Stage 2 proposal](https://github.com/tc39/proposal-js-module-blocks).
+
+<!-- prettier-ignore -->
+```js
+// Input
+module { export let foo = "foo"; };
+
+// Prettier 2.2
+SyntaxError: Unexpected token, expected ";"
+
+// Prettier 2.3
+module {
+  export let foo = "foo";
+};
+```
+
+#### Fix missing parentheses for `yield` in a pipeline ([#10446](https://github.com/prettier/prettier/pull/10446) by [@fisker](https://github.com/fisker))
+
+<!-- prettier-ignore -->
+```jsx
+// Input
+function* f() {
+  return x |> (yield #);
+}
+
+// Prettier 2.2
+function* f() {
+  return x |> yield #;
+}
+
+// Prettier 2.3
+function* f() {
+  return x |> (yield #);
+}
+```
+
+#### Make Babel’s error recovery more selective ([#10495](https://github.com/prettier/prettier/pull/10495) by [@fisker](https://github.com/fisker), [#9787](https://github.com/prettier/prettier/pull/9787) by [@sosukesuzuki](https://github.com/sosukesuzuki), [#10065](https://github.com/prettier/prettier/pull/10065), [#10322](https://github.com/prettier/prettier/pull/10322) by [@thorn0](https://github.com/thorn0))
+
+Previously, because of the [error recovery](https://babeljs.io/blog/2019/11/05/7.7.0#parser-error-recovery-10363httpsgithubcombabelbabelpull10363), the Babel parser was too permissive, which lead to all kinds of AST shapes that Prettier couldn’t print. Prettier 2.3 lets Babel recover only from a few harmless types of errors – for example, multiple `const` declarations with the same name. Anything else is reported as syntax errors.
```

Comment:
"Prettier 2.3" was "Prettier main", replaced by https://github.com/prettier/prettier/blob/4d4c0011b6fd34eea9080a32e6b89e07a90f2f60/scripts/draft-blog-post.js#L142

Should prevent it.

Reply:
This is 100% intended and done in multiple changelog entries.

Reply:
Sorry, I didn't read the whole sentence, I thought it didn't mean `main branch`.